web/src/components/storage/SizeModeSelect.tsx

```diff
@@ -0,0 +1,210 @@
+  );
+}
+
+type CustomSizeProps = {
+  value: SizeRange;
+  onChange: (size: SizeRange) => void;
+};
+
+function CustomSize({ value, onChange }: CustomSizeProps) {
+  const [grow, setGrow] = useState(value.max === NO_VALUE);
+
+  const changeSize = (min: string) => {
+    const max = grow ? NO_VALUE : min;
+    onChange({ min, max });
+  };
+
+  const toggleGrow = () => {
+    const newGrow = !grow;
+    const max = newGrow ? NO_VALUE : value.min;
+    setGrow(newGrow);
+    onChange({ min: value.min, max });
+  };
+
+  const regexp = /^[0-9]+(\.[0-9]+)?(\s*([KkMmGgTtPpEeZzYy][iI]?)?[Bb])$/;
+  const error = value.min && !regexp.test(value.min);
+  const approxMin = error ? null : approxSize(value.min);
+  const help = _(
+    "The size must be a number followed by a unit of the form GiB (power of 2) or GB (power of 10).",
+  );
+
+  if (value.max !== NO_VALUE && value.min !== value.max) {
+    return <UnsupportedSize value={value} onClick={() => changeSize(value.min)} />;
+  }
+
+  return (
+    <Stack hasGutter>
+      <StackItem>
+        <FormGroup fieldId="minSizeValue" label={_("Size")}>
+          <Flex>
+            <FlexItem>
+              <TextInput
+                id="minSizeValue"
+                className="w-14ch"
+                value={value.min}
+                onChange={(_, v) => changeSize(v)}
+              />
+            </FlexItem>
+            {!error && approxMin && (
+              <FlexItem>
+                <SubtleContent>
+                  {
+                    // TRANSLATORS: %s is a disk size (e.g., "10 GiB").
+                    sprintf(_("approx. %s"), approxMin)
+                  }
+                </SubtleContent>
+              </FlexItem>
+            )}
+          </Flex>
+          {!error && <SubtleContent>{help}</SubtleContent>}
+          {error && (
+            <FormHelperText>
+              <HelperText>
+                <HelperTextItem variant="error">{help}</HelperTextItem>
+              </HelperText>
+            </FormHelperText>
+          )}
+        </FormGroup>
+      </StackItem>
+      <StackItem>
+        <Checkbox id="setGrow" label={_("Allow growing")} isChecked={grow} onChange={toggleGrow} />
+        <SubtleContent>
+          {_("The final size can be bigger in order to fill the contiguous free space.")}
+        </SubtleContent>
+      </StackItem>
+    </Stack>
+  );
+}
+
+export type SizeModeSelectProps = {
+  id?: string;
+  value: SizeMode;
+  size: SizeRange;
+  automaticHelp?: React.ReactNode;
+  onChange: (value: SizeMode, size: SizeRange) => void;
+};
+
+export default function SizeModeSelect({
+  id,
+  value,
+  size,
+  onChange,
+  automaticHelp,
+}: SizeModeSelectProps): React.ReactNode {
+  const changeSize = (size: SizeRange) => onChange(value, size);
+
+  return (
+    <Stack hasGutter>
+      <StackItem>
+        <Select
+          id={id || "sizeMode"}
+          value={value}
+          label={value === "auto" ? _("Automatic") : _("Custom")}
```

Comment:
NP: Not sure which word is better now, _Custom_ or _Manual_ (after discarding _Fixed_ because of the "Allow grow" behavior). But let's @ancorgs to decide about final wording :)